src/art/serverless/backend.py

```diff
@@ -167,9 +189,85 @@ async def _train_model(
                 after = event.id
 
     # ------------------------------------------------------------------
-    # Experimental support for S3
+    # Experimental support for S3 and checkpoints
     # ------------------------------------------------------------------
 
+    async def _experimental_pull_model_checkpoint(
+        self,
+        model: "TrainableModel",
+        *,
+        step: int | Literal["latest"] | None = None,
+        local_path: str | None = None,
+        verbose: bool = False,
+        **kwargs: Any,
+    ) -> str:
+        """Pull a model checkpoint from W&B artifacts to a local path.
+
+        For ServerlessBackend, this downloads the checkpoint from W&B artifact storage.
+
+        Args:
+            model: The model to pull checkpoint for.
+            step: The step to pull. Can be an int for a specific step,
+                 or "latest" to pull the latest checkpoint. If None, pulls latest.
+            local_path: Local directory to save the checkpoint. If None, uses temporary directory.
+            verbose: Whether to print verbose output.
+
+        Returns:
+            Path to the local checkpoint directory.
+        """
+        import os
+        import tempfile
+
+        import wandb
+
+        assert model.id is not None, "Model ID is required"
+        assert model.entity is not None, "Model entity is required"
```

Comment:
Would it be hard to pull the user's default entity for this step as well, the same way we do while training?